diem-move/diem-framework/core/sources/DiemConfig.move

```diff
@@ -0,0 +1,152 @@
+    struct Configuration has key {
+        /// Epoch number
+        epoch: u64,
+        /// Time of last reconfiguration. Only changes on reconfiguration events.
+        last_reconfiguration_time: u64,
+        /// Event handle for reconfiguration events
+        events: Event::EventHandle<NewEpochEvent>,
+    }
+
+    /// Reconfiguration disabled if this resource occurs under LibraRoot.
+    struct DisableReconfiguration has key {}
+
+    /// The `Configuration` resource is in an invalid state
+    const ECONFIGURATION: u64 = 0;
+    /// A `DiemConfig` resource is in an invalid state
+    const EDIEM_CONFIG: u64 = 1;
+    /// A `ModifyConfigCapability` is in a different state than was expected
+    const EMODIFY_CAPABILITY: u64 = 2;
+    /// An invalid block time was encountered.
+    const EINVALID_BLOCK_TIME: u64 = 3;
+    /// The largest possible u64 value
+    const MAX_U64: u64 = 18446744073709551615;
+
+    /// Publishes `Configuration` resource. Can only be invoked by Diem root, and only a single time in Genesis.
+    public fun initialize(
+        account: &signer,
+    ) {
+        DiemTimestamp::assert_genesis();
+        SystemAddresses::assert_core_resource(account);
+        assert!(!exists<Configuration>(@CoreResources), Errors::already_published(ECONFIGURATION));
+        move_to<Configuration>(
+            account,
+            Configuration {
+                epoch: 0,
+                last_reconfiguration_time: 0,
+                events: Event::new_event_handle<NewEpochEvent>(account),
+            }
+        );
+    }
+
+    /// Private function to temporarily halt reconfiguration.
+    /// This function should only be used for offline WriteSet generation purpose and should never be invoked on chain.
+    fun disable_reconfiguration(account: &signer) {
+        SystemAddresses::assert_core_resource(account);
+        assert!(reconfiguration_enabled(), Errors::invalid_state(ECONFIGURATION));
+        move_to(account, DisableReconfiguration {} )
+    }
+
+    /// Private function to resume reconfiguration.
+    /// This function should only be used for offline WriteSet generation purpose and should never be invoked on chain.
+    fun enable_reconfiguration(account: &signer) acquires DisableReconfiguration {
+        SystemAddresses::assert_core_resource(account);
+
+        assert!(!reconfiguration_enabled(), Errors::invalid_state(ECONFIGURATION));
+        DisableReconfiguration {} = move_from<DisableReconfiguration>(Signer::address_of(account));
+    }
+
+    fun reconfiguration_enabled(): bool {
+        !exists<DisableReconfiguration>(@CoreResources)
+    }
+
+    /// Signal validators to start using new configuration. Must be called from friend config modules.
+    public(friend) fun reconfigure() acquires Configuration {
+        reconfigure_();
+    }
+
+    /// Signal validators to start using new configuration. Must be called by Diem root.
+    public fun reconfigure_with_core_resource_signer(
```

Comment:
I can't seem to find anywhere where this function is called, and I'm not sure if it should be exposed. What do you think of deleting this function?

Reply:
Ah this function is called by the writeset_epilogue in ExperimentalAccount. Maybe it makes more sense to move writeset_epilogue to basic account?

Reply:
ahh very good point. I think in that case you're right -- moving writeset epilogue to basic account sounds like the right move. I think with that move it would then mean that users of the core configs don't need to think about reconfigurations I believe, which would be nice.